controllers/volumereplicationgroup_controller.go

```diff
@@ -146,6 +158,60 @@ func (r *VolumeReplicationGroupReconciler) configMapFun(configmap client.Object)
 	return req
 }
 
+func (r *VolumeReplicationGroupReconciler) waitForCrds() error {
+	log := r.Log.WithName("checkingDependencies")
+
+	requiredCrds := []string{volumeReplicationClass, volumeReplication}
+
+	for _, crd := range requiredCrds {
+		err := r.waitForResource(log, crd)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *VolumeReplicationGroupReconciler) waitForResource(logger logr.Logger, resourceName string) error {
+	unstructuredResource := &unstructured.UnstructuredList{}
+	unstructuredResource.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   volrep.GroupVersion.Group,
+		Kind:    resourceName,
+		Version: volrep.GroupVersion.Version,
+	})
+
+	err := r.Client.List(context.TODO(), unstructuredResource)
+	if err != nil {
+		if !meta.IsNoMatchError(err) {
+			logger.Error(err, "unexpected error occurred while checking CRD", "CRD", resourceName)
+
+			return err
+		}
+
+		logger.Info(fmt.Sprintf("CRD %s is not installed on cluster, waiting until it appears", resourceName))
```

Comment:
Our logging infrastructure do no support formatting? Formatting logs in the logger is very basic and important for performance - we pay for formatting only if we actually log the string.

Reply:
If I want to build log message in specific format - crd name inside the message - this is the only way. It is possible to use another approach - message and then resource name in the end - we can write something like this: 
log.Info("CRD is not installed on cluster", "resource name", resourceName)
I just wanted to build message in specific order. Also it happens once, not in loop, so we have no performance issue here

Reply:
It is true that in this loop formatting a message that will not be logged is not an issue. The issue is once you do it here people will copy this pattern to places where it matters. Also we don't want to have special logging code in some cases to avoid performance issues, we want the same logging pattern everywhere that is efficient enough to be used everywhere. But looks like our logging does not support formatting so unless we replace the logger we cannot do much about this.

Reply:
I am confused, what do you want to see here? Please, explain

Reply:
If our logger does not support formatting, we have no other choice so this if fine.